pkg/broker/job_state_subscriber.go

```diff
@@ -0,0 +1,106 @@
+package broker
+
+import (
+	"fmt"
+
+	"github.com/openshift/ansible-service-broker/pkg/apb"
+)
+
+// JobStateSubscriber is responsible for handling and persisiting JobState changes
+type JobStateSubscriber struct {
+	dao SubscriberDAO
+}
+
+// NewJobStateSubscriber returns a newly initialised JobStateSubscriber
+func NewJobStateSubscriber(dao SubscriberDAO) *JobStateSubscriber {
+	return &JobStateSubscriber{
+		dao: dao,
+	}
+}
+
+func isBinding(msg JobMsg) bool {
+	return msg.State.Method == apb.JobMethodBind || msg.State.Method == apb.JobMethodUnbind
+}
+
+// ID is used as an identifier for the type of subscriber
+func (jss *JobStateSubscriber) ID() string {
+	return "jobstate"
+}
+
+// Notify external API to notify this subscriber of a change in the Job
+func (jss *JobStateSubscriber) Notify(msg JobMsg) {
+	log.Debugf("JobStateSubscriber Notify : msg state %v ", msg.State)
+	id := msg.InstanceUUID
+	if isBinding(msg) {
+		id = msg.BindingUUID
```

Comment:
Can `msg.BindingUUID` ever be `""`? I would probably change `isBinding` to also ensure `BindingUUID` is not empty. Otherwise, we would store an empty string as the key. Maybe there is no way this could happen, I'll see how the flow goes.

Reply:
It seems if it can ever be empty that it should fail before getting here, similar with instanceUUID.